cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -0,0 +1,314 @@
+    original_roles = server_info.server_roles_db
+    server_info.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor')
+    server_info.disable_server_roles('automate', 'smartstate')
+    command = ('Metric::Targets.perf_capture_always = {:storage=>true, :host_and_cluster=>true};')
+    appliance.ssh_client.run_rails_command(command, timeout=None)
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    command = ('Metric::Targets.perf_capture_always = {:storage=>false, :host_and_cluster=>false};')
+    appliance.ssh_client.run_rails_command(command, timeout=None)
+
+
+def verify_records_rollups_table(appliance, provider):
+    # Verify that hourly rollups are present in the metric_rollups table.
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        if (record.cpu_usagemhz_rate_average or
+           record.cpu_usage_rate_average or
+           record.derived_memory_used or
+           record.net_usage_rate_average or
+           record.disk_usage_rate_average):
+            return True
+    return False
+
+
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
+    # Retrieve resource usage values from metric_rollups table.
+    average_cpu_used_in_mhz = 0
+    average_memory_used_in_mb = 0
+    average_network_io = 0
+    average_disk_io = 0
+
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+    metrics = appliance.db.client['metrics']
+    rollups = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+    logger.info('Deleting METRICS DATA from metrics and metric_rollups tables')
+
+    appliance.db.client.session.query(metrics).delete()
+    appliance.db.client.session.query(rollups).delete()
+
+    # Chargeback reporting is done on hourly and daily rollup values and not real-time values.So, we
+    # are capturing C&U data and forcing hourly rollups by running these commands through
+    # the Rails console.
+
+    def verify_records_metrics_table(appliance, provider):
+        # Verify that rollups are present in the metric_rollups table.
+        vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+        ems = appliance.db.client['ext_management_systems']
+        metrics = appliance.db.client['metrics']
+
+        rc, out = appliance.ssh_client.run_rails_command(
```

Comment:
**Required**

Use the SSHResult object that's returned, its built-in method overrides allow it to be directly asserted against for an rc==0 comparison, or its stdout if used as a string.

Reply:
Done